cou/steps/hypervisor.py

```diff
@@ -29,23 +31,49 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass(frozen=True)
-class HypervisorMachine:
-    """Hypervisor machine containing units for multiple applications."""
-
-    name: str
-    units: list[COUUnit]
-
-
-@dataclass(frozen=True)
+@dataclass
 class HypervisorGroup:
     """Group of hypervisors.
 
     For example, this can represent a group of all hypervisors in a single availability zone.
     """
 
     name: str
-    hypervisors: list[HypervisorMachine]
+    apps: dict[str, list[COUUnit]]
+
+    def __eq__(self, other: Any) -> bool:
+        """Equal magic method for HypervisorGroup.
+
+        :param other: HypervisorGroup object to compare.
+        :type other: Any
+        :return: True if equal False if different.
+        :rtype: bool
+        """
+        if not isinstance(other, HypervisorGroup):
+            return NotImplemented
+
+        return other.name == self.name
+
+
+class AZs(defaultdict):
+    """AZs dictionary object with default value HypervisorGroup."""
+
+    def __init__(self) -> None:
+        """Initialize the Hypervisor class.
+
+        The AZs represent default dict, with predefined HypervisorGroup as default value.
+        """
+        super().__init__(default_factory=None)
```

Comment:
This is not needed since we are overriding `__missing__` method (where `default_factory` was normally used)

Reply:
It is because without this you can do `AZs(default_factory=MyFactory)`, which will be ignored.